### PR TITLE
feat(dli): add new datasource to get list of spark templates

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**:
   add new datasource to get list of spark templates.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/dli" TESTARGS="-run TestAccDataSourceDliSparkTemplates_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run TestAccDataSourceDliSparkTemplates_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDliSparkTemplates_basic
=== PAUSE TestAccDataSourceDliSparkTemplates_basic
=== CONT  TestAccDataSourceDliSparkTemplates_basic
--- PASS: TestAccDataSourceDliSparkTemplates_basic (42.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       42.270s
```

### CODE REVIEW

File: docs/data-sources/dli_spark_templates.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  The [body](#templates_body_struct) structure is documented below.
+
+* `group` - The group of template.
+
+<a name="templates_body_struct"></a>
+The `body` block supports:
+
+* `app_parameters` - The input parameters of the main class, that is application parameters.
+
+* `name` - The Spark job name.
```
Comment (deer-hang) [inline docs/data-sources/dli_spark_templates.md]: Change word lowercase. Check the other parts.

File: docs/data-sources/dli_spark_templates.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+* `obs_bucket` - The OBS bucket for storing the Spark jobs.
+
+* `auto_recovery` - The whether to enable the retry function.
```
Comment (deer-hang) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `auto_recovery` - Indicates whether to enable the retry function.
```

File: docs/data-sources/dli_spark_templates.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+* `configurations` - The configuration items of the DLI Spark.
+
+* `executor_memory` - The executor memory of the Spark application.
+
+* `queue_name` - The DLI queue name.
+
+* `main_class` - The Spark main class of the template.
+
+* `python_files` - The name of the resource package of type pyFile upload to the DLI resource management system.
```
Comment (deer-hang) [inline docs/data-sources/dli_spark_templates.md]: Please refer to the documentation to modify the globally ambiguous fields.

https://support.huaweicloud.com/intl/en-us/api-dli/dli_02_0354.html

File: huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+}
+
+func dataSourceDliSparkTemplatesRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	wrapper := newSparkTemplatesDSWrapper(d, meta)
+	lisSpaJobTemRst, err := wrapper.ListSparkJobTemplates()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, _ := uuid.GenerateUUID()
```
Comment (deer-hang) [inline huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go]: The error message returned by the function cannot be ignored.

File: docs/data-sources/dli_spark_templates.md (hunk 1 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 2 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 3 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go (hunk 4 above)
Comment (wangyuancheng01) [reply huaweicloud/services/dli/data_source_huaweicloud_dli_spark_templates.go]: done

File: docs/data-sources/dli_spark_templates.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  The [templates](#templates_struct) structure is documented below.
+
+<a name="templates_struct"></a>
+The `templates` block supports:
+
+* `id` - The ID of template.
+
+* `name` - The name of template.
+
+* `group` - The group name to which the spark templates belong.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]:
```suggestion
* `group` - The group name to which the spark template belongs.
```

File: docs/data-sources/dli_spark_templates.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+* `name` - The name of resource.
+
+* `type` - The type of resource.
+
+<a name="body_resources_struct"></a>
+The `resources` block supports:
+
+* `name` - The name of resource.
+
+* `type` - The type of resource.
```
Comment (Lance52259) [inline docs/data-sources/dli_spark_templates.md]: The same structure definition does not need to be declared twice.
The same H5 tag can be referenced.

File: docs/data-sources/dli_spark_templates.md (hunk 5 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

File: docs/data-sources/dli_spark_templates.md (hunk 6 above)
Comment (wangyuancheng01) [reply docs/data-sources/dli_spark_templates.md]: done

**Pull request merged**